contracts/L1GlobalForkRequester.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity ^0.8.20;
+
+/* 
+   Contract to proxy a fork request from the bridge to its ForkingManager
+   Any L2 contract can call us over the bridge to get us to request a fork.
+   We record the dispute they were forking over
+*/
+
+// TODO: An alternative would be to look up the ForkingManager on L2ChainInfo and call it directly.
+// TODO: We could use this to manage the dispute data, in which case we could gate initiateFork() to only be callable by us.
+// TODO: The whitepaper implies a whitelist about who is allowed to fork us. Currently anybody can as long as they pay.
+
+import {IForkingManager} from "./interfaces/IForkingManager.sol";
+import {IPolygonZkEVMBridge} from "@RealityETH/zkevm-contracts/contracts/interfaces/IPolygonZkEVMBridge.sol";
+import {IPolygonZkEVM} from "@RealityETH/zkevm-contracts/contracts/interfaces/IPolygonZkEVM.sol";
+
+// NB We'd normally use the interface IForkableBridge here but it causes an error:
+//  Error (5005): Linearization of inheritance graph impossible
+import {ForkableBridge} from "./ForkableBridge.sol";
+
+contract L1GlobalForkRequester {
+
+    // Any bridge (or any contract pretending to be a bridge) can call this.
+    // We'll look up its ForkingManager and ask it for a fork.
+    // TODO: It might make more sense if this contract requested the chain ID then kept a record of it.
+    // ...then the ForkingManager would be locked to only fork based on our request.
+
+    function onMessageReceived(address _originAddress, uint32 _originNetwork, bytes memory _data) external payable {
+
+      ForkableBridge bridge = ForkableBridge(msg.sender);
+      IForkingManager fm = IForkingManager(bridge.forkmanager());
```

Comment:
i think code is much easier to read if we write out the names: fm-> forkingManager

Reply:
I don't have a strong opinion but I generally find it easier to read short names for local variables that only have a short life, and leave the grander ones for instance variables.

Reply:
my opinion is:
- writing longer names has basically no disadvantage, as IDEs do autocomplete.
- when reading code, sometimes my shortterm memory is not good enough to keep all variable names present, hence, its easier to identify them if the words are written out. I think its common best practice